src/panels/config/devices/ha-config-devices-dashboard.ts

```diff
@@ -0,0 +1,173 @@
+  RowClickedEvent,
+} from "../../../components/ha-data-table";
+// tslint:disable-next-line
+import { DeviceRegistryEntry } from "../../../data/device_registry";
+import { EntityRegistryEntry } from "../../../data/entity_registry";
+import { ConfigEntry } from "../../../data/config_entries";
+import { AreaRegistryEntry } from "../../../data/area_registry";
+import { navigate } from "../../../common/navigate";
+import { HassEntity } from "home-assistant-js-websocket";
+
+@customElement("ha-config-devices-dashboard")
+export class HaConfigDeviceDashboard extends LitElement {
+  @property() public hass!: HomeAssistant;
+  @property() public devices!: DeviceRegistryEntry[];
+  @property() public entries!: ConfigEntry[];
+  @property() public entities!: EntityRegistryEntry[];
+  @property() public areas!: AreaRegistryEntry[];
+  @property() public domain!: string;
+
+  private _devices = memoizeOne(
+    (
+      devices: DeviceRegistryEntry[],
+      entries: ConfigEntry[],
+      domain: string
+    ) => {
+      if (!domain) {
+        return devices;
+      }
+      return devices.filter(
+        (device) =>
+          entries.find((entry) =>
+            device.config_entries.includes(entry.entry_id)
+          )!.domain === domain
+      );
+    }
+  );
+
+  private _columns: DataTabelColumnContainer = {
+    device_name: {
+      title: "Device",
+      sortable: true,
+      filterable: true,
+      direction: "asc",
+    },
+    manufacturer: {
+      title: "Manufacturer",
+      sortable: true,
+      filterable: true,
+    },
+    model: {
+      title: "Model",
+      sortable: true,
+      filterable: true,
+    },
+    area: {
+      title: "Area",
+      sortable: true,
+      filterable: true,
+    },
+    integration: {
+      title: "Integration",
+      sortable: true,
+      filterable: true,
+    },
+    battery: {
+      title: "Battery",
+      sortable: true,
+      filterable: true,
+      type: "numeric",
+      template: (battery: HassEntity) =>
+        battery
+          ? html`
+              ${battery.state}%
+              <ha-state-icon
+                .hass=${this.hass!}
+                .stateObj=${battery}
+              ></ha-state-icon>
+            `
+          : html`
+              n/a
+            `,
+    },
+  };
+
+  protected render(): TemplateResult {
+    return html`
+      <hass-subpage
+        header=${this.hass.localize("ui.panel.config.devices.caption")}
+      >
+        <ha-data-table
+          .columns=${this._columns}
+          .data=${this._devices(this.devices, this.entries, this.domain).map(
+            (device) => {
+              return {
+                device_name: device.name_by_user || device.name,
+                id: device.id,
+                manufacturer: device.manufacturer,
+                model: device.model,
+                area:
+                  !this.areas || !device || !device.area_id
+                    ? "No area"
+                    : this.areas.find(
```

Comment:
I suggest that our memoization function create a data structure that gives us devices, but also already finds areas, integrations and entity ID representing the battery status (but not the actual fetching of state, as we should not memoize anything that requires `hass.states`, as that breaks caching).